index.js

```diff
@@ -48,21 +50,47 @@ export default async function pMap(
 				return;
 			}
 
-			resolvingCount++;
+			pendingIndexes.add(index);
 
 			(async () => {
 				try {
 					const element = await nextItem.value;
+
+					if (finished) {
+						return;
+					}
+
 					result[index] = await mapper(element, index);
```

Comment:
Edge case: if the `pMap` iteration is stopped while an element of the original array is being awaited for, I believe the mapping should not even start. This way, this small part of the diff is actually performing a bug fix. Should I move this into a separate PR?

(previously, the only situation that could cause this was an error (with `stopOnError` set to `true`), but with this PR a call to `.stop()` is also possible)

Reply:
> Should I move this into a separate PR?

That would be great 👍

Reply:
Please see #40